astropy/io/fits/connect.py

```diff
@@ -251,16 +258,23 @@ def read_table_fits(
                 astropy_native=astropy_native,
                 unit_parse_strict=unit_parse_strict,
                 mask_invalid=mask_invalid,
+                strip_spaces=strip_spaces,
             )
         finally:
             hdulist.close()
 
-    # In the loop below we access the data using data[col.name] rather than
-    # col.array to make sure that the data is scaled correctly if needed.
     data = table.data
 
     columns = []
     for col in data.columns:
+        # use data[col.name] rather than col.array to make sure that the data
+        # is scaled correctly if needed.
+        arr = data[col.name]
+        coltype = col.dtype.subdtype[0].type if col.dtype.subdtype else col.dtype.type
```

Comment:
I wondered whether this was needed and one could just do `if strip_spaces and arr.dtype.kind in "SU"`, but I now see that it is used further down so really no point, at least for this PR.